cdap-support-bundle/src/main/java/io/cdap/cdap/support/services/SupportBundleGenerator.java

```diff
@@ -193,4 +242,28 @@ private NamespaceId validNamespace(NamespaceId namespace) throws Exception {
 
     return namespace;
   }
+
+  private void collectSupportBundleTaskStatus(SupportBundlePipelineStatus supportBundlePipelineStatus,
+                                              SupportBundleTaskStatus supportBundleTaskStatus) {
+    int lastIndexOfDot = supportBundleTaskStatus.getType().lastIndexOf(".");
+    SupportBundleTaskType taskType =
+      SupportBundleTaskType.valueOf(supportBundleTaskStatus.getType().substring(lastIndexOfDot + 1));
```

Comment:
This kind of name manipulation is hard to maintain. Please refactor it into proper method(s) in the enum(s) for conversion.